src/crypto.rs

```diff
@@ -1,73 +1,63 @@
+    while {
+        let read_size = stream.read(&mut buffer)?;
+        hash.update(&buffer[..read_size]);
+        read_size == BUFFER_SIZE
+    } {}
     Ok(format!("{:x}", hash.finalize()))
 }
 
-/// Function to hash data that is already in memory. This is much cleaner and less overhead compare
-/// to mapping data in memory into a ``BufReader``
-pub fn hash_bytes(bytes: impl AsRef<[u8]>) -> String {
-    format!("{:x}", Sha256::digest(bytes.as_ref()))
+/// Evaluate checksum hash of raw data in memory.
+pub fn hash_buffer(buffer: impl AsRef<[u8]>) -> String {
+    format!("{:x}", Sha256::digest(buffer.as_ref()))
 }
 
-/// Compute the checksum of the content of a folder or file
+/// Evaluate checksum hash of a file.
 ///
 /// # Errors
-/// Will error out if accessing the files fails
-pub fn compute_checksum_for_path(path: &dyn AsRef<Path>) -> Result<String> {
-    let unref_path = path.as_ref();
-    if !unref_path.exists() {
-        return Err(OrcaError::from(Kind::PathDoesNotExist {
-            path: unref_path.to_path_buf(),
-        }));
-    }
-
-    if unref_path.is_file() {
-        // Read and hash in chunks
-        hash_stream(&mut File::open(unref_path)?)
-    } else if unref_path.is_dir() {
-        // Path is a directory, thus we will need to recursively hash and sort
-        let hashes = unref_path
-            .read_dir()?
-            .map(|dir_entry| compute_checksum_for_path(&dir_entry?.path()))
-            .collect::<Result<BTreeSet<String>>>()?;
-
-        // Combine all the hashes by alpha numeric order
-        let mut hashes_buffer = String::new();
-        for hash in hashes {
-            hashes_buffer.push_str(&hash);
-        }
+///
+/// Will return error if unable to access file.
+pub fn hash_file(filepath: impl AsRef<Path>) -> Result<String> {
+    hash_stream(&mut File::open(filepath)?)
+}
 
-        // Hash the buffer
-        Ok(hash_bytes(hashes_buffer))
-    } else {
-        // Unknown type of path or unsupported, thus panic for now
-        Err(OrcaError::from(Kind::UnsupportedPath {
-            path: unref_path.to_path_buf(),
-        }))
-    }
+/// Evaluate checksum hash of a folder.
+///
+/// # Errors
+///
+/// Will return error if unable to access any child file.
+pub fn hash_dir(dirpath: impl AsRef<Path>) -> Result<String> {
+    let summary: BTreeMap<String, String> = dirpath
+        .as_ref()
+        .read_dir()?
+        .map(|path| {
+            let access_path = path?.path();
+            let hash_path = access_path
+                .strip_prefix(&dirpath)?
+                .to_string_lossy()
+                .into_owned();
+            Ok((
+                hash_path,
+                if access_path.is_dir() {
+                    hash_dir(access_path)?
+                } else {
+                    hash_file(access_path)?
+                },
+            ))
+        })
+        .collect::<Result<_>>()?;
+    Ok(hash_buffer(serde_yaml::to_string(&summary)?))
```

Comment:
I am not sure about including relative path into a BTreeMap then converting it to yaml only to be trashed since hashing compress all of that.

Perhaps something like this would be a hybrid approach of our versions

```
    let summary: BTreeSet<String> = dirpath
        .as_ref()
        .read_dir()?
        .map(|path| {
            let access_path = path?.path();
            Ok((if access_path.is_dir() {
                hash_dir(access_path)?
            } else {
                hash_file(access_path)?
            },))
        })
        .collect::<Result<_>>()?;
    Ok(hash_buffer(serde_yaml::to_string(&summary)?))
```